treeherder/perf/perf_sheriff_bot.py

```diff
@@ -15,14 +23,133 @@ class PerfSheriffBot:
     Wrapper class used to aggregate the reporting of backfill reports.
     """
 
-    def __init__(self, report_maintainer):
+    DEFAULT_MAX_RUNTIME = timedelta(minutes=50)
+
+    def __init__(
+        self,
+        report_maintainer: BackfillReportMaintainer,
+        backfill_tool: BackfillTool,
+        secretary_tool: SecretaryTool,
+        max_runtime: timedelta = None,
+        backfill_tool_disabled: bool = True,
+        secretary_tool_disabled: bool = True,
+    ):
         self.report_maintainer = report_maintainer
+        self.backfill_tool = backfill_tool
+        self.secretary = secretary_tool
+        self._woke_up_time = datetime.now()
+        self._max_runtime = self.DEFAULT_MAX_RUNTIME if max_runtime is None else max_runtime
+
+        # TODO: feature flags; remove when enabled ->
+        self.__backfill_tool_disabled = backfill_tool_disabled
+        self.__secretary_tool_disabled = secretary_tool_disabled
+
+        # extra precautions
+        if self.__backfill_tool_disabled:
+            self.backfill_tool = None
+        if self.__secretary_tool_disabled:
+            self.secretary = None
+        # -> up to here
+
+    def sheriff(self, since: datetime, frameworks: List[str], repositories: List[str]):
+        self.assert_can_run()
+
+        logger.info(f'{self.__class__.__name__} now performs sheriff duty.')
```

Comment:
What's the purpose of this log line (what does "now performs sheriff duty" mean)?

Reply:
It's mostly for locating the log lines of this bot. This line marks the moment right after the bot starts doing its job (generating reports, backfilling, checking outcome of old backfills).

Reply:
We're trying to remove log lines unless they're useful. Can we rely on success or failure messages instead for debugging purposes instead?

Reply:
Yes, we could.